File: packages/qa_native/macos/Classes/qa_audio_device_forwarder.c
// Forwarder for the device layer — same pattern as the others: CocoaPods
// podspecs cannot reference files outside their own directory.
//
// Separate from the engine and decoder forwarders because all three are
// separate translation units everywhere else; miniaudio alone is ~96k
// lines and has no business sharing an object file with the hot loops.
#include "../../src/qa_audio_device.c"

File: packages/qa_native/ios/Classes/qa_audio_device_forwarder.m
// Objective-C (.m), NOT .c — deliberately.
//
// miniaudio's Apple backend includes <AVFoundation/AVFoundation.h> on iOS
// (it drives AVAudioSession there). AVFoundation is an Objective-C module,
// so a plain .c translation unit fails to build it outright:
//
//   Parse Issue (Xcode): Could not build module 'AVFoundation'
//
// The file is otherwise the same forwarder pattern as the others —
// CocoaPods podspecs cannot reference files outside their own directory.
// `s.source_files = 'Classes/**/*'` picks up .m as readily as .c, and
// OTHER_CFLAGS (including the -ffp-contract=off parity flag) applies to
// Objective-C exactly as it does to C.
#include "../../src/qa_audio_device.c"
